### ISSUE TITLE
`ActivityThread.currentApplication()` is null since 1.2.0

### ISSUE THREAD

**Issue by benoitletondor** [cashapp/paparazzi#689]
**Description**
I have some snapshots that are using `HtmlCompat.fromHtml` which uses `ActivityThread.currentApplication()` when it sees a `</b>` tag (in the `handleEndTag` method of `Html.java`). This was working before and now it crashes because `ActivityThread.currentApplication()` is null.

**Steps to Reproduce**
Try to snapshot a view which renders HTML with a `<b>` tag:

```
val spanned = HtmlCompat.fromHtml("This is a <b>bold</b> tag", HtmlCompat.FROM_HTML_MODE_COMPACT)
textView.text = spanned
```

**Expected behavior**
It was working great before, now it's crashing, so I'm expecting this to work as before 1.2.0 (1.1.0 was working great)

**Additional information:**
- Paparazzi Version: 1.2.0
- OS: Android 33, macOS 13.1, OpenJDK Runtime Environment Temurin-17.0.5+8 (build 17.0.5+8)
- Compile SDK: 33
- Gradle Version: 7.6
- Android Gradle Plugin Version: 7.4

**Comment by TWiStErRob** [comment]
Repro project and stack trace in #694

**Comment by saket** [comment]
It looks like `Html` needs access to the `Application` instance [starting from Android 33](https://cs.android.com/android/platform/superproject/+/master:frameworks/base/core/java/android/text/Html.java;l=861;drc=0a674a8a4433aa1fcaca5bcaec53e4f8004e8a6a), which wasn't the [case earlier](https://cs.android.com/android/platform/superproject/+/android-12.0.0_r34:frameworks/base/core/java/android/text/Html.java;l=863;drc=5d123b67756dffcfdebdb936ab2de2b29c799321). I'll investigate if we can support this. 

In the meantime, you can unblock yourself by pinning your test to Android 32:

```kotlin
val paparazzi = Paparazzi(
  environment = detectEnvironment().copy(
    platformDir = "${androidHome()}/platforms/android-32",
  )
)
```

**Comment by ronnieotieno** [comment]
> It looks like `Html` needs access to the `Application` instance [starting from Android 33](https://cs.android.com/android/platform/superproject/+/master:frameworks/base/core/java/android/text/Html.java;l=861;drc=0a674a8a4433aa1fcaca5bcaec53e4f8004e8a6a), which wasn't the [case earlier](https://cs.android.com/android/platform/superproject/+/android-12.0.0_r34:frameworks/base/core/java/android/text/Html.java;l=863;drc=5d123b67756dffcfdebdb936ab2de2b29c799321). I'll investigate if we can support this.
> 
> In the meantime, you can unblock yourself by pinning your test to Android 32:
> 
> ```kotlin
> val paparazzi = Paparazzi(
>   environment = detectEnvironment().copy(
>     platformDir = "${androidHome()}/platforms/android-32",
>   )
> )
> ```

Did this work for you @benoitletondor? I just tried, and even reduced the compiled version of the entire app from 33 to 32 and still not working.

**Comment by benoitletondor** [comment]
Hey @saket and @ronnieotieno .

Sorry I didn't reply earlier, I don't think it's related to the API level indeed as it was working with the same setup targetting API 33 using https://github.com/chrisbanes/paparazzi 

I feel like this is another change that now doesn't provide a proper Application for the `ActivityThread` to use, but I have to admit I didn't dig deeper.

My workaround has just been to not use `<b>` tags in my tests HTML, which is find for my use case.

**Comment by saket** [comment]
> Did this work for you @benoitletondor? I just tried, and even reduced the compiled version of the entire app from 33 to 32 and still not working.

Whoops I'm sorry, setting paparazzi's `platformDir` will only affect resources and is insufficient to workaround this crash.

> I don't think it's related to the API level indeed as it was working with the same setup targetting API 33 using https://github.com/chrisbanes/paparazzi

@chrisbanes can confirm this, but from what I understand, his fork was still using an older version of layoutlib that doesn't render views against API 33. 

I was also able to verify that layoutlib runs into the same problem. Android Studio Flamingo Canary 10 fails to preview layouts that use `Html.fromHtml` with `<b>` tags:

![Screenshot 2023-01-25 at 1 20 12 PM](https://user-images.githubusercontent.com/2387680/214688186-1a1fc4c1-6647-4d48-b4a8-3c26596480a1.png)

I've filed an issue for this: https://issuetracker.google.com/u/1/issues/266681481

**Comment by chrisbanes** [comment]
> but from what I understand, his fork was still using an older version of layoutlib that doesn't render views against API 33.

Confirmed. Each layoutlib version is tested against whatever SDK versions available at the time, and the layoutlib used in 1.1.x was quite old (it only worked up to SDK 32 iirc).

My fork basically 'enforced' that and declared the SDK_INT to match, allowing apps to avoid SDK 33+ APIs.

Anyway, this looks like a layoutlib issue, as @saket said
